Zend/zend_compile.c

```diff
@@ -8306,6 +8306,12 @@ static zend_op_array *zend_compile_func_decl_ex(
 
 	if (decl->kind == ZEND_AST_CLOSURE || decl->kind == ZEND_AST_ARROW_FUNC) {
 		op_array->fn_flags |= ZEND_ACC_CLOSURE;
+		/* Set the closure scope at compile time as an optimization to
```

Comment:
Is https://github.com/php/php-src/issues/10782 issue related and possibly fixed as well?